weather/router.py

```diff
@@ -0,0 +1,46 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+from dependencies import get_db
+from weather import schemas, crud
+
+
+router = APIRouter()
+
+
+@router.get("/cities/", response_model=list[schemas.City])
+async def read_all_cities(db: Session = Depends(get_db)):
```

Comment:
The `db` parameter is defined as a `Session`, but the CRUD operations expect an `AsyncSession`. Ensure that `get_db` provides an `AsyncSession` to avoid runtime errors.